tests/pastefromoffice.js

```diff
@@ -3,87 +3,116 @@
-	let editor, content, normalizeSpy;
+describe( 'PasteFromOffice', () => {
+	let editor, pasteFromOffice, clipboard;
 
 	testUtils.createSinonSandbox();
 
-	before( () => {
-		content = new DocumentFragment();
-	} );
-
 	beforeEach( () => {
-		return VirtualTestEditor
-			.create( {
-				plugins: [ Clipboard, PasteFromOffice ]
-			} )
-			.then( newEditor => {
-				editor = newEditor;
-				normalizeSpy = testUtils.sinon.spy( editor.plugins.get( 'PasteFromOffice' ), '_normalizeWordInput' );
+		return VirtualTestEditor.create( {
+			plugins: [ PasteFromOffice ]
+		} )
+			.then( _editor => {
+				editor = _editor;
+				pasteFromOffice = editor.plugins.get( 'PasteFromOffice' );
+				clipboard = editor.plugins.get( 'Clipboard' );
 			} );
 	} );
 
-	it( 'runs normalizations if Word meta tag detected #1', () => {
-		const dataTransfer = createDataTransfer( {
-			'text/html': '<meta name=Generator content="Microsoft Word 15">'
-		} );
-
-		editor.plugins.get( 'Clipboard' ).fire( 'inputTransformation', { content, dataTransfer } );
-
-		expect( normalizeSpy.calledOnce ).to.true;
+	it( 'is Paste from Office', () => {
+		expect( pasteFromOffice ).to.be.instanceOf( PasteFromOffice );
 	} );
 
-	it( 'runs normalizations if Word meta tag detected #2', () => {
-		const dataTransfer = createDataTransfer( {
-			'text/html': '<html><head><meta name="Generator"  content=Microsoft Word 15></head></html>'
-		} );
-
-		editor.plugins.get( 'Clipboard' ).fire( 'inputTransformation', { content, dataTransfer } );
-
-		expect( normalizeSpy.calledOnce ).to.true;
+	it( 'should have static name', () => {
+		expect( PasteFromOffice.pluginName ).to.equal( 'PasteFromOffice' );
 	} );
 
-	it( 'does not normalize the content without Word meta tag', () => {
-		const dataTransfer = createDataTransfer( {
-			'text/html': '<meta name=Generator content="Other">'
-		} );
-
-		editor.plugins.get( 'Clipboard' ).fire( 'inputTransformation', { content, dataTransfer } );
-
-		expect( normalizeSpy.called ).to.false;
+	it( 'should load Clipboard plugin', () => {
+		expect( editor.plugins.get( Clipboard ) ).to.be.instanceOf( Clipboard );
 	} );
 
-	it( 'does not process content many times for the same `inputTransformation` event', () => {
-		const clipboard = editor.plugins.get( 'Clipboard' );
-
-		const dataTransfer = createDataTransfer( {
-			'text/html': '<html><head><meta name="Generator"  content=Microsoft Word 15></head></html>'
+	describe( 'isTransformedWithPasteFromOffice - flag', () => {
+		const htmlDataProcessor = new HtmlDataProcessor();
+
+		describe( 'data which should be processed', () => {
+			[
+				{
+					'text/html': '<html><head><meta name="Generator"  content=Microsoft Word 15></head></html>'
+				},
+				{
+					'text/html': '<meta name=Generator content="Microsoft Word 15">'
+				},
+				{
+					'text/html': '<p id="docs-internal-guid-12345678-1234-1234-1234-1234567890ab"></p>'
+				}
+			].forEach( ( inputData, index ) => {
```

Comment:
In most cases, we're creating a function and call it with different parameters. Also, the `text/html` is redundant here as it is always the same so we can rewrite it to something like this:

```js

testInput( '<html>foo</html>', 'MS Word - case 1' );

testInput( '<html>bar</html>', 'MS Word - case 2' );

testInput( '<html>baz</html>', 'Google Docs' );
```

The test name would also have more meaning then `case #1` as you cannot easily search for failed test in the file.